src/components/composer/composer/composer.ts

```diff
@@ -302,17 +302,27 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
     this.compositionActive = false;
   }
 
+  private updateSelectionIfRequired(ev: KeyboardEvent) {
+    const { start, end } = this.contentHelper.getCurrentSelection();
+    this.env.model.dispatch("CHANGE_COMPOSER_CURSOR_SELECTION", { start, end });
+    this.isKeyStillDown = true;
+  }
+
   onKeydown(ev: KeyboardEvent) {
     let handler = this.keyMapping[ev.key];
     if (handler) {
       handler.call(this, ev);
+      if (
+        this.env.model.getters.getEditionMode() === "editing" &&
+        !(["Enter", "Tab"].includes(ev.key) && !(ev.altKey || ev.ctrlKey))
```

Comment:
while the code changed, the situation raised by Lucas is still present, you even managed to make it worse...  I am confused because I don't know where the issue lies in our explanation, I will try to be more clear.

Lucas said: 
> What I don't like about the proposed solution here is that handlers business are leaking in the generic onKeydown.

this means that what we would like to get rid of is: " the mention of `Enter` or `Tab`  because these notions of which key was pressed are handled in the [**handlers**](https://github.com/odoo/o-spreadsheet/commit/d71874993cb17d71ff972e208a536c195b7b4b31#diff-ea6fa27863d7bc4c77db3f9c449298a78c2e689d06ed2d92f0e3207fc4025251R312),  not in the general `onKeydown`  function.  your current proposition does not change that at all.

We want  `updateSelectionRequired`  to be only called by specific handlers that require it - so in `keymapping`, some entries would handle it, others would not.

Aside from this,  there is no point in forseeing the upcoming multine behaviour with 
```javascript
!(ev.altKey || ev.ctrlKey)
```
1. multi line is not there yet, so this accomplishes nothing
2. this only applies to `Enter`, not `Tab`
3.  so again, this hints that it should be handled separately in the handlers, and not in the generic caller `onKeyDown`